lib/types.nix

```diff
@@ -137,47 +137,60 @@ rec {
-          default = null;
-          description = ''
-            Lua code inserted at the start of the plugin's configuration.
-            This is the same as using `lib.nixvim.utils.mkBeforeSection` when defining `content`.
-          '';
-        };
-        post = lib.mkOption {
-          type = with types; nullOr lines;
-          default = null;
-          description = ''
-            Lua code inserted at the end of the plugin's configuration.
-            This is the same as using `lib.nixvim.utils.mkAfterSection` when defining `content`.
-          '';
-        };
-        content = lib.mkOption {
-          type = types.lines;
-          default = "";
-          description = ''
-            Configuration of the plugin.
+  pluginLuaConfig =
+    { hasContent }:
+    types.submodule (
+      { config, ... }:
+      let
+        inherit (builtins) toString;
+        inherit (lib.nixvim.utils) mkBeforeSection mkAfterSection;
+      in
+      {
+        options =
+          {
+            pre = lib.mkOption {
+              type = with types; nullOr lines;
+              default = null;
+              description =
+                ''
+                  Lua code inserted at the start of the plugin's configuration.
+                ''
+                + lib.optionalString hasContent ''
+                  This is the same as using `lib.nixvim.utils.mkBeforeSection` when defining `content`.
+                '';
+            };
+            post = lib.mkOption {
+              type = with types; nullOr lines;
+              default = null;
+              description =
+                ''
+                  Lua code inserted at the end of the plugin's configuration.
+                ''
+                + lib.optionalString hasContent ''
+                  This is the same as using `lib.nixvim.utils.mkAfterSection` when defining `content`.
+                '';
+            };
+          }
+          // (lib.optionalAttrs hasContent) {
+            content = lib.mkOption {
+              type = types.lines;
+              default = "";
+              description = ''
+                Configuration of the plugin.
 
-            If `pre` and/or `post` are non-null, they will be merged using the order priorities
-            ${toString (mkBeforeSection null).priority} and ${toString (mkBeforeSection null).priority}
-            respectively.
-          '';
-        };
-      };
+                If `pre` and/or `post` are non-null, they will be merged using the order priorities
+                ${toString (mkBeforeSection null).priority} and ${toString (mkBeforeSection null).priority}
+                respectively.
+              '';
+            };
+          };
 
-      config.content = lib.mkMerge (
-        lib.optional (config.pre != null) (mkBeforeSection config.pre)
-        ++ lib.optional (config.post != null) (mkAfterSection config.post)
-      );
-    }
-  );
+        config = lib.optionalAttrs hasContent {
```

Comment:
This feels like it'd be simpler to have two modules, e.g. using `types.submoduleWith`

```nix
types.submoduleWith {
  modules =
    [
      {/* mod1 has common options */}
    ]
    ++ lib.optionals hasContent [
      {/*mod2 has content option */}
      ({config, ...}: {/* mod3 has config definitions*/})
    ];
};
```

Or maybe using a let-in?
```nix
let
  commonModule = {};
  configOptionsModule = {};
  configMergingModule = {config, ...}: {};
in
types.submoduleWith {
  modules =
    [ commonModule ]
    ++ lib.optionals hasContent [
      configOptionsModule
      configMergingModule
    ];
};
```